src/components/FileList.tsx

```diff
@@ -197,7 +242,12 @@ export class FileList extends React.Component<IFileListProps, IFileListState> {
   /** Handle right-click on an untracked file */
   contextMenuUntracked = (event: React.MouseEvent) => {
     event.preventDefault();
-    this._contextMenuUntracked.open(event.clientX, event.clientY);
+    const extension = PathExt.extname(this.state.selectedFile.to);
+    if (extension.length > 0 && extension !== 'ipynb' && extension !== 'py') {
+      this._contextMenuUntracked.open(event.clientX, event.clientY);
+    } else {
+      this._contextMenuUntrackedMin.open(event.clientX, event.clientY);
+    }
```

Comment:
I agree with @kgryte. We should not forbid ignoring `ipynb` or `py` files. In particular for the notebook extension, Jupytext users may want to ignore all ipynb files.

But I understand the necessity to protect newbies against that. So what about asking a confirmation for those use cases? With some warning message like:

```ts
Are you sure you want to ignore <b>all ${extension} files</b> within this git repository?
```

> The best place for such confirmation is in the command directly.

Reply:
I have added a dialog box to ask confirmation when user wants to ignore a file extension.